test/bluemix.test.js

```diff
@@ -37,193 +37,222 @@ function itSkipIf(flag) {
 }
 
 describe('loopback:bluemix generator', function() {
+  this.timeout(30 * 60 * 1000); // 30 minutes
   beforeEach(common.resetWorkspace);
+  beforeEach(function createSandbox(done) {
+    helpers.testDirectory(SANDBOX, done);
+  });
 
-  beforeEach(function(done) {
-    fs.ensureDir(SANDBOX, function() {
-      process.chdir(SANDBOX);
-      common.createDummyProject(SANDBOX, 'test-app', done);
-    });
+  beforeEach(function createProject(done) {
+    common.createDummyProject(SANDBOX, 'test-app', done);
   });
 
   afterEach(function(done) {
     process.chdir('/');
     rimraf(SANDBOX, done);
   });
 
-  it('should generate datasources.bluemix.js', function(done) {
-    var gen = givenBluemixGenerator('--force --bluemix');
-    helpers.mockPrompt(gen, {
-      enableManifest: false,
-    });
-    gen.run(function() {
-      ygAssert.file('./server/datasources.bluemix.js');
-      done();
-    });
+  it('should generate datasources.bluemix.js', function() {
+    return helpers.run(path.join(__dirname, '../bluemix'))
+      .cd(SANDBOX)
+      .withPrompts({
+        enableManifest: false,
+      })
+      .withOptions({
+        force: true,
```

Comment:
why do we need these options?

Reply:
The old generator is created by calling [`givenBluemixGenerator('--force --bluemix');`](https://github.com/strongloop/generator-loopback/blob/master/test/bluemix.test.js#L55), that's where the options set :)